src/components/Tables/PowerSummaryTable.js

```diff
@@ -17,91 +17,308 @@ function PowerSummaryTableToolTip({ title, statusColor }) {
     </Tooltip>
   );
 }
+
 function PowerSummaryTable({
-  title, data, total, percent,
+  title, data = [], total = 0, percent = 0, deviceId = 'MPW1',
 }) {
-  function getErrors(messages) {
-    if (messages === undefined) return [];
-    const errors = messages.filter((item) => item.filter((inner) => inner.type === 'error').length > 0);
-    return errors;
-  }
-  function getWarning(messages) {
-    if (messages === undefined) return [];
-    const warnings = messages.filter((item) => item.filter((inner) => inner.type === 'warn').length > 0);
-    return warnings;
-  }
-  function buildMessage(messages) {
-    return messages.reduce((sum, item, currentIndex) => {
-      item.forEach((i, index) => sum.push(
-        // eslint-disable-next-line react/no-array-index-key
-        <span key={`${currentIndex}+${index}`}>
-          {i.text}
-          <br />
-        </span>,
-      ));
-      return sum;
-    }, []);
-  }
-  function message(messages) {
-    const errors = getErrors(messages);
-    if (errors.length > 0) {
-      return buildMessage(errors);
+  const [thermalData, setThermalData] = useState({
+    ambientTypical: 25,
+    ambientWorstCase: 50,
+    thetaJa: 10,
+  });
+
+  const [powerData, setPowerData] = useState({
+    powerBudget: 1.0,
+    fpgaScaling: 25,
+    pcScaling: 25,
+  });
+
+  const ambientTypicalRef = useRef(null);
+  const ambientWorstCaseRef = useRef(null);
+  const thetaJaRef = useRef(null);
+  const powerBudgetRef = useRef(null);
+  const fpgaScalingRef = useRef(null);
+  const pcScalingRef = useRef(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        // Update the API call to match the correct endpoint
+        GET(deviceInfo(deviceId), (result) => {
+          if (result && result.specification) {
+            const { specification } = result;
+
+            // Process thermal data
+            setThermalData({
+              ambientTypical: specification.thermal?.ambient?.typical || 25,
+              ambientWorstCase: specification.thermal?.ambient?.worstcase || 50,
```

Comment:
This is not the correct prop 'worstcase'. The correct prop is 'worsecase'.

Reply:
@chernyee please pull the latest updates and let me know if this is ok, lets connect on this on my monday evening so we can push the changes once everything looks good.